src/Icicle/Source/Checker/Constraint.hs

```diff
@@ -174,7 +174,8 @@ generateQ qq@(Query (c:_) _)
             retKey     <- Temporality TemporalityElement . TypeVar <$> fresh
             retVal     <- Temporality TemporalityElement . TypeVar <$> fresh
             (q',sq,t') <- withBind k retKey
-                        $ withBind v retVal rest
+                        $ withBind v retVal
+                        $ removeElementBinds >> rest
```

Comment:
oh, right. `withBind` binds the name, then run the action. I need to `removeElementBinds` before all this.